server/routes/getNewProfile.ts

```diff
@@ -0,0 +1,85 @@
+import { Request, Response, NextFunction } from 'express';
+import { Op } from 'sequelize';
+import { factory, formatFilename } from '../../shared/logging';
+const log = factory.getLogger(formatFilename(__filename));
+import { DB } from '../database';
+
+export const Errors = {
+  NoAddressProvided: 'No address provided in query',
+  NoAddressFound: 'No address found',
+  NoProfileFound: 'No profile found',
+};
+
+const getNewProfile = async (models: DB, req: Request, res: Response, next: NextFunction) => {
+  const { address } = req.query;
+  if (!address) return next(new Error(Errors.NoAddressProvided));
+
+  const addressModel = await models.Address.findOne({
+    where: {
+      address,
```

Comment:
We should distinguish by chain as well as just address, as there may be duplicates across chains.

Reply:
This is why we're deduplicating the db. Making the duplicates an issue upstream of this.

Reply:
Also, if the encoding is downstream of a private key, regardless of the address "found" by this call, shouldn't any of the collisions also be owned by the same user? Then when all addresses are found in line 32, it's fine?

Reply:
Just tacking on here that `chain` was not included as a query parameter for this route. My understanding was that data deduplication would be handled separately/outside of this endpoint.

Reply:
@zakhap are we expecting this PR to function properly prior to address deduplication? Or should we wait to merge until that goes in?